multiversx_sdk_cli/transactions.py

```diff
@@ -323,3 +119,90 @@ def get_guardian_account_from_args(args: Any):
         raise errors.NoWalletProvided()
 
     return account
+
+
+def send_and_wait_for_result(transaction: ITransaction, proxy: INetworkProvider, timeout: int) -> ITransactionOnNetwork:
+    if not transaction.signature:
+        raise errors.TransactionIsNotSigned()
+
+    txOnNetwork = _send_transaction_and_wait_for_result(proxy, transaction, timeout)
+    return txOnNetwork
+
+
+def _send_transaction_and_wait_for_result(proxy: INetworkProvider, payload: ITransaction, num_seconds_timeout: int = 100) -> ITransactionOnNetwork:
+    AWAIT_TRANSACTION_PERIOD = 5
+
+    tx_hash = proxy.send_transaction(payload)
+    num_periods_to_wait = int(num_seconds_timeout / AWAIT_TRANSACTION_PERIOD)
+
+    for _ in range(0, num_periods_to_wait):
+        time.sleep(AWAIT_TRANSACTION_PERIOD)
+
+        tx = proxy.get_transaction(tx_hash)
+        if tx.is_completed:
+            return tx
+        else:
+            logger.info("Transaction not yet done.")
+
+    raise errors.KnownError("Took too long to get transaction.")
+
+
+def tx_to_dictionary_as_inner(tx: Transaction) -> Dict[str, Any]:
+    dictionary = tx.to_dictionary()
+    dictionary["receiver"] = base64.b64encode(bytes.fromhex(tx.receiver.hex())).decode()  # type: ignore
+    dictionary["sender"] = base64.b64encode(bytes.fromhex(tx.sender.hex())).decode()  # type: ignore
+    dictionary["chainID"] = base64.b64encode(tx.chainID.encode()).decode()
+    dictionary["signature"] = base64.b64encode(bytes(bytearray(tx.signature))).decode()
+    dictionary["value"] = tx.value
+
+    return dictionary
+
+
+def _dict_to_json(dictionary: Dict[str, Any]) -> bytes:
+    serialized = json.dumps(dictionary, separators=(',', ':')).encode("utf8")
+    return serialized
+
+
+def compute_relayed_v1_data(tx: Transaction) -> str:
+    inner_dictionary = tx_to_dictionary_as_inner(tx)
+    serialized = _dict_to_json(inner_dictionary)
+    serialized_hex = serialized.hex()
+    return f"relayedTx@{serialized_hex}"
+
+
+def load_transaction_from_file(f: TextIO) -> Transaction:
+    data_json: bytes = f.read().encode()
+    fields = json.loads(data_json).get("tx") or json.loads(data_json).get("emittedTransaction")
+
+    instance = JSONTransaction()
+    instance.__dict__.update(fields)
```

Comment:
Perhaps we could have used `Transaction.from_dictionary`?

https://github.com/multiversx/mx-sdk-py-core/blob/main/multiversx_sdk_core/transaction.py#L105

Then, we can remove `JSONTransaction` as well, I think. 

(for a future PR).

Reply:
Indeed. I forgot we have that in sdk-core. Will be replaced in a future PR.